src/utils/streams.js

```diff
@@ -0,0 +1,177 @@
+		console.error("Convert from file requries valid file to work with");
+		return;
+	}
+	const csv = require("csvjson");
+	const content = fs.readFileSync(filePath, "utf8");
+	process.stdout.write(util.format(csv.toObject(content)));
+}
+
+/**
+ * Creates new json file from the content of the parameter at the location of the source file
+ */
+function convertToFile(filePath) {
+	if (!fs.statSync(filePath).isFile()) {
+		console.error("Convert to file requries valid file to work with");
+		return;
+	}
+
+	const csv = require("csvjson");
+	const content = fs.readFileSync(filePath, "utf8");
+	const parsedPath = path.parse(filePath);
+	const targetPath = path.join(parsedPath.dir, parsedPath.name + ".json");
+
+	const outStream = fs.createWriteStream(targetPath)
+	outStream.write(JSON.stringify(csv.toObject(content)));
+	outStream.end();
+}
+
+
+function cssBundler(directory) {
+	if (!fs.statSync(directory).isDirectory()) {
+		console.error("Css bundler requires valid directory to work with");
+		return;
+	}
+	fs.readdir(directory, (err, files) => {
+		if (err) {
+			console.error(err);
+			return;
+		}
+		files = files
+			.filter(file => file.endsWith(".css"))
+			.filter(file => file !== "bundle.css")
+			.map(file => path.join(directory, file));
+		//append required file to the end
+		var homeworkCss = path.join(__dirname, "/data/nodejs-homework3.css");
+		files.push(homeworkCss)
+
+		console.log(`Bundling css files: ${files}`);
+
+		const targetcssPath = path.join(directory, "bundle.css");
+
+		async.map(files, fs.readFile, (err, results) => {
+			if (err) {
+				console.error(err);
+				return;
+			}
+
+			fs.writeFile(targetcssPath, results.join("\n"), (err) => {
+				if (err) {
+					console.error(err);
+					return;
+				}
+			});
+
+		});
+	});
+
+}
+
+
+function registerAction(action, parameter) {
+	actionRegistry[action.name] = () => action(parameter);
+}
+
+/**
+ * Writes defined input, or in it's absence a dynamic promp input to process std through a transformer function
+ */
+function inputTransform(prompt, transformer, definedInput) {
+	if (definedInput) {
+		process.stdout.write(transformer(definedInput));
+	}
+	else {
+		var readline = require('readline');
+		var rl = readline.createInterface(process.stdin, process.stdout);
+		rl.setPrompt(`${prompt}> `);
+		rl.prompt();
+		rl.on('line', function (line) {
+			if (line == "exit") {
+				console.log("bye");
+				rl.close();
+			}
+			console.log(transformer(line));
+			rl.prompt();
+		}).on('close', function () {
+			process.exit(0);
+		});
+	}
+}
+
+registerAction(reverse, argv._[0]);
+registerAction(transform, argv._[0]);
+registerAction(outputFile, argv.file);
+registerAction(convertFromFile, argv.file);
+registerAction(convertToFile, argv.file);
+registerAction(cssBundler, argv.path);
+
+
+if (process.argv[2] && (process.argv[2] == "-h" || process.argv[2] == "--help")) {
+	yargs.showHelp();
+}
+else {
+	if (actionRegistry[argv.action]) {
+		actionRegistry[argv.action]();
```

Comment:
some error handling seem to be missing, what if value was not provided, value was provided but in not expected format, e.g. file does not exist, file exists but folder was expected, folder exists but file is expected.